module/project/templates/helper.js

```diff
@@ -1079,6 +1079,18 @@ module.exports = class ProjectTemplatesHelper {
                         userId : userId,
                         projectTemplateId : templateData[0]._id
                     },["_id"]);
+
+                    //check for program enrollment if enrolled then add programJoined true
+                    let programJoined = await programUsers.findProgramIds(userId)
```

Comment:
@ankitpws  I don't get the logic here pls correct if I am wrong. you are trying to check if the user joined program of the project right? But if you query only with userId it will return all userJoined programs right? Why can't we query with both userId and programId and check if data is present or not?